modules/core/include/opencv2/core.hpp

```diff
@@ -819,12 +819,31 @@ mixChannels , or split .
 @param minLoc pointer to the returned minimum location (in 2D case); NULL is used if not required.
 @param maxLoc pointer to the returned maximum location (in 2D case); NULL is used if not required.
 @param mask optional mask used to select a sub-array.
-@sa max, min, compare, inRange, extractImageCOI, mixChannels, split, Mat::reshape
+@sa max, min, argMin, argMax, compare, inRange, extractImageCOI, mixChannels, split, Mat::reshape
 */
 CV_EXPORTS_W void minMaxLoc(InputArray src, CV_OUT double* minVal,
                             CV_OUT double* maxVal = 0, CV_OUT Point* minLoc = 0,
                             CV_OUT Point* maxLoc = 0, InputArray mask = noArray());
 
+/**
+ * @brief Finds indices of max elements along provided axis
+ * @param src input single-channel array
+ * @param dst output array
+ * @param axis axis
+ * @param select_last_index indicates whether to choose the first maximum or the last if there are many
+ * @sa argMin, minMaxLoc, min, max, compare
+ */
+CV_EXPORTS_W void argMax(InputArray src, OutputArray dst, int axis = 0, bool select_last_index = false);
```

Comment:
BTW, there is [`reduce()`](https://docs.opencv.org/3.4/d2/de8/group__core__array.html#ga4b78072a303f29d9031d56e5638da78e) call in OpenCV for similar operations.

Reply:
Maybe it makes sense to add REDUCE_{MIN,MAX}_INDEX to reduce itself along with support for higher dimensions.